### ISSUE TITLE
Inclusion of "id_token_hint" for keycloak logout and other generic_oauth clients in deprecation of  previous "redirect_url"

### ISSUE THREAD

**Issue by VisheshIrasus** [grafana/grafana#67987]
### 
# What went wrong?

**What happened**:

  - Keycloak has updated the logout mechanism as mentioned at https://www.keycloak.org/docs/latest/securing_apps/#logout . 

**What did you expect to happen**:

  - .

### 
# How do we reproduce it?

**Step 1**:

- Setup the following in the grafana.in.
`signout_redirect_url =https://${KEYCLOAK_SERVICE}.${HOST_SERVER}:${HTTPS_PORT}/realms/${REALM}/protocol/openid-connect/logout?redirect_uri=https://${SERVICE_NAME}.${HOST_SERVER}:${HTTPS_PORT}/logout
**Step 2**:
-You should get invalid redirect uri

### 
# What Grafana version are you using?

v8.2.6

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

None

### User's OS?

Ubuntu latest

### Is this a Regression?

None

### Anything else to add?

So, to fix the issue as mentioned in keycloak docs.
https://www.keycloak.org/docs/latest/securing_apps/#logout

I tried searching the source code and looks like, merely storing the token as the environment variable should solve the problem. 
But, I don't know about what security loopholes it might open.
I want to locate the `/grafana/pkg/api/pluginproxy/token_provider_generic.go` file inside the docker image which I 'm unable to .

My idea is to add the following code:

```go
// Store the access token as an environment variable
err = os.Setenv("GF_GENERIC_OAUTH_TOKEN_FROM_OPENID_CLIENT", token.AccessToken)
if err != nil {
	return "", err
}
```
to the function `func (provider *genericAccessTokenProvider) GetAccessToken() (string, error) `
Thereafter user can use the following in the grafana.ini:

```ini

signout_redirect_url =https://${KEYCLOAK_SERVICE}.${HOST_SERVER}:${HTTPS_PORT}/realms/${REALM}/protocol/openid-connect/logout?post_logout_redirect_uri=https://${SERVICE_NAME}.${HOST_SERVER}:${HTTPS_PORT}/logout&id_token_hint=${GF_AUTH_GENERIC_OAUTH_TOKEN}
```
```[tasklist]
### Tasks
```

**Comment by VisheshIrasus** [comment]
Also I want to expose the API to the user, the realm, and client id and secret have to be hard coded currently

**Comment by usmangt** [comment]
:Hi @VisheshIrasus,

Thanks for opening this issue.

I will forward this to our Engineering team so that they can have a look and provide more information.

**Comment by peytonyip** [comment]
> I Just found a mitigation of this in keycloak document. Please pass the client_id and post_logout_redirect_uri parameter with grafana logout url. it will work

Hi,I just started using KeyCloak. Can you provide a specific configuration example?
I look from https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/keycloak/#enable-single-logout ，but the url in Note is 404

**Comment by VisheshIrasus** [comment]
Read the keycloak docs. In the grafana.ini file, set the mentioned link in
the keycloak docs for logout.

**Comment by Jguer** [comment]
The documentation on this section is outdated and is fixed in https://github.com/grafana/grafana/pull/79890. 

Grafana now adds the id_token_hint when oauth is configured as of Grafana 10.2 https://github.com/grafana/grafana/pull/70357

**Comment by klaask** [comment]
> The documentation on this section is outdated and is fixed in [#79890](https://github.com/grafana/grafana/pull/79890).
> 
> Grafana now adds the id_token_hint when oauth is configured as of Grafana 10.2 [#70357](https://github.com/grafana/grafana/pull/70357)

Grafana adds the id_token_hint only if the client_id parameter exists. This should be added to the documentation. Right now keycloak responds with 400 status code, if you configure signout_redirect_url this way:

`signout_redirect_url = https://<PROVIDER_DOMAIN>/realms/<REALM_NAME>/protocol/openid-connect/logout?post_logout_redirect_uri=https%3A%2F%2F<GRAFANA_DOMAIN>%2Flogin
`

Logout works if you add the client_id parameter:

`signout_redirect_url = https://<PROVIDER_DOMAIN>/realms/<REALM_NAME>/protocol/openid-connect/logout?post_logout_redirect_uri=https%3A%2F%2F<GRAFANA_DOMAIN>%2Flogin&client_id=<CLIENT_ID>
`
